### DISCUSSION TITLE
Add new device class for binary sensors:  tamper

### DISCUSSION THREAD

**Discussion by frenck** [home-assistant/architecture#642] [Entity Models] [answered]
# Context

Some sensors provide an indicator / can detect if they are being tampered with.
This is generally fairly common on devices that are related to security/alarm systems. Examples include things like keypads and motion sensors.

Some integration that can provide these classes/tampering sensors: Z-Wave JS, Zigbee (ZHA/deCONZ/Zigbee2MQTT) and Tuya (the latter caused this proposal).

# Proposal

Add new device class `DEVICE_CLASS_TAMPER` to binary sensor platform, to indicate a device has been tampered with, is detecting tampering, or has detected tampering.

The visual icons in the front end would be the same as the problem device class.

# Consequences

Another device class? 🤷 

Implementation reference:

- https://github.com/home-assistant/core/pull/57632
- https://github.com/home-assistant/frontend/pull/10268
- https://github.com/home-assistant/developers.home-assistant/pull/1084
- https://github.com/home-assistant/home-assistant.io/pull/19755

**Comment by balloob** [comment]
Sounds good :+1:

**Comment by MartinHjelmare** [comment]
Yes, sounds good.

**Comment by Kane610** [comment]
Looks good!

**Comment by Mariusthvdb** [comment]
is this only for the binary Frenck? asking because flood sensors by Fibaro have 2 related sensors:
`sensor.<name>_flood_sensor_home_security_cover_status`
`binary_sensor.<name>_flood_sensor_home_security_tampering_product_cover_removed`

**Reply by frenck** [reply]
Yes only binary sensor.

**Comment by frenck** [comment, marked as answer]
This device class has been implemented and made available. Both user and developer documentation has been updated.
